cmd/mayactl/app/command/volume_info.go

```diff
@@ -0,0 +1,177 @@
+	collection := client.ReplicaCollection{}
+	controllerClient := client.ControllerClient{}
+	_, err = controllerClient.GetVolumeStats(annotation.ClusterIP+v1.ControllerPort, v1.InfoAPI, &collection)
+	if err != nil {
+		return err
+	}
+
+	c.DisplayVolumeInfo(annotation, collection)
+	return nil
+}
+
+// DisplayVolumeInfo displays the outputs in standard I/O.
+// Currently It displays volume access modes and target portal details only.
+func (c *CmdVolumeInfoOptions) DisplayVolumeInfo(a *Annotations, collection client.ReplicaCollection) error {
+	var (
+		// address, mode are used here as blackbox for the replica info
+		// address keeps the ip and access mode details respectively.
+		address, mode []string
+		replicaCount  int
+		portalInfo    PortalInfo
+	)
+	const (
+		replicaTemplate = `
+================= Replica Details =================
+IP            AccessMode               Status
+{{range $key, $value := .}}
+{{$value.IP}}     {{$value.AccessMode}}                       {{$value.Status}}
+{{end}}
+===================================================
+`
+		portalTemplate = `
+================= Portal Details ==================
+IQN     :   {{.IQN}}
+Volume  :   {{.VolumeName}}
+Portal  :   {{.Portal}}
+Size    :   {{.Size}}
+Status  :   {{.Status}}
+===================================================
+`
+	)
+	portalInfo = PortalInfo{
+		a.Iqn,
+		c.volName,
+		a.TargetPortal,
+		a.VolSize,
+		a.ControllerStatus,
+	}
+	tmpl, err := template.New("VolumeInfo").Parse(portalTemplate)
+	if err != nil {
+		fmt.Println("Can't Parse the template, found error :", err)
+		return nil
+	}
+	err = tmpl.Execute(os.Stdout, portalInfo)
+	if err != nil {
+		fmt.Println("Can't execute the template, found error :", err)
+		return nil
+	}
+	replicaCount, _ = strconv.Atoi(a.ReplicaCount)
+	length := len(collection.Data)
+	// This case will occur only if user has manually specified zero replica.
+	if replicaCount == 0 || length == 0 {
+		fmt.Println("No Replicas are running, please check the volume's pod specification or try again later.")
+		return nil
+	}
+	// We get the info of the running replicas from the collection.data.
+	// If there are no replicas running they are either in CrashedLoopBackOff
+	// or in Pending or in ImagePullBackoff.In such cases it will show Waiting
+	// NA,  NA in Status, access mode and IP fields respectively.
+	replicaInfo := make(map[int]*ReplicaInfo)
+	for key, _ := range collection.Data {
+		address = append(address, strings.TrimSuffix(strings.TrimPrefix(collection.Data[key].Address, "tcp://"), v1.ReplicaPort))
+		mode = append(mode, collection.Data[key].Mode)
+		replicaInfo[key] = &ReplicaInfo{address[key], mode[key], "Running"}
+	}
+	if length < replicaCount {
+		for i := length; i < (replicaCount); i++ {
+			replicaInfo[i] = &ReplicaInfo{"NA", "       NA", "Waiting"}
+		}
+	}
+	tmpl = template.New("ReplicaInfo")
+	tmpl = template.Must(tmpl.Parse(replicaTemplate))
+	err = tmpl.Execute(os.Stdout, replicaInfo)
```

Comment:
should this error me caught and displayed as error message? "Unable to display replica details"

Reply:
Thanks for pointing this, will update it soon.